src/pattern-library/components/patterns/ColorFoundations.js

```diff
@@ -1,74 +1,48 @@
+import tailwindConfig from '../../../../tailwind.config';
+
 import Library from '../Library';
 
 // TODO:
 // - color descriptions and guidance
-// - hex and other metadata about colors
 // - foreground color examples
-// - more examples of overriding with `!` classes
 // - valid contrast combinations
 
-const brand = ['brand'];
-
-const greys = [
-  'white',
-  'grey-0',
-  'grey-1',
-  'grey-2',
-  'grey-3',
-  'grey-4',
-  'grey-5',
-  'grey-6',
-  'grey-7',
-  'grey-8',
-  'grey-9',
-];
-
-const states = ['success', 'notice', 'error'];
+const colorConfig = tailwindConfig.theme.extend.colors ?? {};
 
-function ColorSwatch({ colorName }) {
+function ColorSwatch({ colorKey }) {
+  const colorValue =
+    typeof colorConfig[colorKey] === 'string'
+      ? `: ${colorConfig[colorKey]}`
+      : '';
   return (
     <div>
-      <div
-        className={`hyp-u-bg-color--${colorName}`}
-        style="width:16rem;height:2rem;margin-right:1em"
-      />
+      <div className={`bg-${colorKey} h-12 mr-4`} />
       <p>
-        <i>{colorName}</i>
+        <i>
+          {colorKey}
+          {colorValue}
+        </i>
       </p>
     </div>
   );
 }
 
-const brandExamples = brand.map(colorName => (
-  <ColorSwatch key={colorName} colorName={colorName} />
-));
-
-const greyExamples = greys.map(colorName => (
-  <ColorSwatch key={colorName} colorName={colorName} />
-));
-
-const stateExamples = states.map(colorName => (
-  <ColorSwatch key={colorName} colorName={colorName} />
+const colorSwatches = Object.keys(colorConfig).map(colorKey => (
+  <ColorSwatch key={colorKey} colorKey={colorKey} />
 ));
 
 export default function ColorFoundations() {
   return (
     <Library.Page title="Colors">
-      <Library.Pattern title="Brand red">
-        <div className="hyp-u-layout-row" style="flex-wrap:wrap">
-          {brandExamples}
-        </div>
-      </Library.Pattern>
-
-      <Library.Pattern title="Greys">
-        <div className="hyp-u-layout-row" style="flex-wrap:wrap">
-          {greyExamples}
+      <Library.Pattern title="Color swatches">
+        <div className="grid md:grid-cols-2 lg:grid-cols-3 gap-4">
```

Comment:
This concise bit of utility usage, boom, gives us a three-column grid on wide screens, 2 on mid, 1 on narrow, with appropriate gaps, without me needing to author and test widely. Yep, we can define the breakpoints (and/or rename them) in our config.

Reply:
I should also point out that we can use an option in tailwind to prefix all tailwind classes if we want.